migrations/migration_test.go

```diff
@@ -2761,3 +2763,325 @@ func TestFixLoadedBrokenReferences(t *testing.T) {
 	err = storage.CheckHealth()
 	require.NoError(t, err)
 }
+
+// TestMigrateNestedValue is a reproducer for issue #3288.
```

Comment:
This was rushed out before breakfast.  I'm adding one more test to this PR before our meeting.